specification/network/resource-manager/Microsoft.Network/preview/2022-06-01-preview/networkManagerGroupMembership.json

```diff
@@ -0,0 +1,252 @@
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v4/types.json#/definitions/Resource"
+        }
+      ],
+      "description": "Resource of networkGroupMembership."
+    },
+    "NetworkGroupMembershipProperties": {
+      "type": "object",
+      "properties": {
+        "groupMemberships": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/NetworkGroupMembershipGroup"
+          },
+          "description": "GroupMembership list"
+        }
+      },
+      "description": "Network Group Membership Properties."
+    },
+    "NetworkGroupMembershipGroup": {
+      "type": "object",
+      "properties": {
+        "networkGroupId": {
+          "type": "string",
+          "description": "network Group ID."
+        },
+        "sources": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/BaseNetworkGroupMembershipSource"
+          },
+          "description": "Source list"
+        }
+      },
+      "description": "The network manager effective connectivity configuration"
+    },
+    "BaseNetworkGroupMembershipSource": {
+      "type": "object",
+      "properties": {
+        "kind": {
+          "type": "string",
+          "description": "Whether the NetworkGroupMembershipSource is StaticMembership or Policy.",
+          "enum": [
+            "StaticMembership",
+            "Policy"
+          ],
+          "x-ms-enum": {
+            "name": "Type",
+            "modelAsString": true
+          }
+        }
+      },
+      "required": [
+        "kind"
+      ],
+      "discriminator": "kind",
+      "description": "Base Network Group Membership Source."
+    },
+    "StaticNetworkGroupMembershipSource": {
+      "type": "object",
+      "properties": {
+        "staticMemberId": {
+          "type": "string",
+          "description": "Static Member Id."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/BaseNetworkGroupMembershipSource"
+        }
+      ],
+      "description": "Static Network Group Membership Source.",
+      "x-ms-discriminator-value": "StaticMembership"
+    },
+    "PolicyNetworkGroupMembershipSource": {
+      "type": "object",
+      "properties": {
+        "policyAssignmentId": {
+          "type": "string",
+          "description": "Policy Assignment Id."
+        },
+        "policySetDefinitionId": {
+          "type": "string",
+          "description": "Policy Set Definition Id."
+        },
+        "policyDefinitionId": {
+          "type": "string",
+          "description": "Policy Definition Id."
+        },
+        "policyDefinitionReferenceId": {
+          "type": "string",
+          "description": "Policy Definition Reference Id."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/BaseNetworkGroupMembershipSource"
+        }
+      ],
+      "description": "Policy Membership Source.",
```

Comment:
'Policy Network Group Membership Source.' if we want to be consistent with static description